src/hash.hpp

```diff
@@ -24,6 +26,40 @@ namespace Simhash {
      * @return hash representative of the content of the text */
     template <typename Hash=jenkins>
     class Simhash {
+    private:
+        class Accumulator {
+        private:
+            int64_t v[64];
+
+            size_t window;
+            Cyclic<hash_t> cyclic;
+
+        public:
+            Accumulator(): window(3), cyclic(window) {
+                memset(v, 0, 64);
+            }
+
+            void update(hash_t hash) {
+                int j(0);
+                for (j = 63; j > 0; --j) {
```

Comment:
Also, this loop being `j > 0` is a bit deceptive. Normally it should be `j >= 0` if you count down. I might actually be tempted to make it `j >= 1` just to make it more confirm what it is doing. That's also why it was fine that that was actually an unsigned loop variable before, but it is still strange.